dbms/src/Databases/DatabaseFactory.cpp

```diff
@@ -2,29 +2,76 @@
 #include <Databases/DatabaseOrdinary.h>
 #include <Databases/DatabaseMemory.h>
 #include <Databases/DatabaseDictionary.h>
+#include <Parsers/ASTLiteral.h>
+#include <Parsers/formatAST.h>
+#include <Common/parseAddress.h>
+#include <Common/config.h>
+
+#if USE_MYSQL
+
+#include <Databases/DatabaseMySQL.h>
+
+#endif
+
 
 namespace DB
 {
 
 namespace ErrorCodes
 {
-    extern const int UNKNOWN_DATABASE_ENGINE;
+extern const int BAD_ARGUMENTS;
+extern const int UNKNOWN_DATABASE_ENGINE;
 }
 
-
 DatabasePtr DatabaseFactory::get(
-    const String & engine_name,
     const String & database_name,
     const String & metadata_path,
+    const ASTStorage * engine_define,
     Context & context)
 {
+    String engine_name = engine_define->engine->name;
+
+    if (engine_name != "MySQL")
+    {
+        if (engine_define->engine->arguments || engine_define->engine->parameters || engine_define->partition_by ||
+            engine_define->primary_key || engine_define->order_by || engine_define->sample_by || engine_define->settings)
+        {
+            std::stringstream ostr;
+            formatAST(*engine_define, ostr, false, false);
+            throw Exception("Unknown database engine: " + ostr.str(), ErrorCodes::UNKNOWN_DATABASE_ENGINE);
```

Comment:
You can make error message more descriptive, like "Database engine XYZ cannot have arguments".

PS. And `parameters`, `primary_key`, `order_by`, `sample_by`, `settings` should not ever be in CREATE DATABASE statement.